### ISSUE TITLE
Allow graph an tag count, fix display errors

### ISSUE THREAD

**Issue by github-pba** [TheHive-Project/TheHive#1038]
### Request Type
Feature Request / Enhancement

### Work Environment

| Question              | Answer
|---------------------------|--------------------
| OS version (server)       |  Ubuntu
| OS version (client)       | Win7
| TheHive version / git hash   | 3.4.0-RC3
| Package Type              |  DEB
 Browser type & version    | any

## Trying to count specific tags fails
I want to count the occurences of specific tags (not _cases containing_ a specific tag). The tags are "Team A", "Team B" and "Team C". The purpose is counting team performance (yes, I know, I should'nt count this way, but hey...).

To do this, I create a Donut graph with the following definitions:
![Def Page 1](https://user-images.githubusercontent.com/50981940/60655975-f5ccfb00-9e4e-11e9-97b0-0c53875bbfeb.png)
![Def Page 2](https://user-images.githubusercontent.com/50981940/60655980-f8c7eb80-9e4e-11e9-8832-b547e7d9ae77.png)
![Def Page 3](https://user-images.githubusercontent.com/50981940/60655991-fcf40900-9e4e-11e9-8685-414c4fbdbcd4.png)

The result I get is this. It does not only show the tags I filtered on, but a new (I'll explain the reason for the problem in a minute):
![Donut Graph](https://user-images.githubusercontent.com/50981940/60656405-a6d39580-9e4f-11e9-9f7b-a7a616991d3f.png)

My records are:

| Case | Tags
|---|--------------------
| 1 |  Team A, disturbing-tag
| 2 |  Team B, disturbing-tag
| 3 |  Team A, disturbing-tag
| 4 |  Team C, disturbing-tag

The problem here is that every case has, in addition to the Team A/B/C tag another tag, named "disturbing-tag".

So internally, TheHive filters on the tags, and returns four records. Then, it starts to count the occurences of the tags, with the result:

| Tag | count 
|-----|-----
| Team A | 2
| Team B | 1
| Team C | 1
| disturbing-tag | 4

Now it sorts the list and applies the limit of 3, which returns 

| Tag | count 
|-----|-----
| disturbing-tag | 4
| Team A | 2
| Team B | 1

As long as the total number of occurences of a tag is larger than ANY of the count of the tags I filter on, it will push itself into the graph.

As a conclusion, you can't define a graph on the count of the occurence of a tag, because TheHive does not count tags, only cases.

**So, finally, the feature request is to have the possibility to draw a pie graph on the count of a defined set of tags.**

## wrong default sort order
In the sort dialog, there is no sort order preselected, but internally the default sort is "descending". The selector should be pre-selected to descending to reflect whats actually happening.

## page size depends on wrong HTML object
The scroll bar for the page depends on the frame with the row objects (which, in turn, holds the graphs). 

When working on a small screen (1366 x 768 in my case), the lower end of the graph object list is off-screen and can't be scrolled into view:
![Object offscreen](https://user-images.githubusercontent.com/50981940/60654972-44799580-9e4d-11e9-946b-c11c6fa98bf0.png)

The scroll bar should refer to either the list of rows, or the size of the toolbar, which ever is bigger.

**Comment by nadouani** [comment]
I think that what you want to achieve requires another type of widget, not just an aggregation donut.

We could solve this using a donut that allows defining series, so we can think of three series in your case, one for each team tag.

I was working on something like this, but it's not yet complete.

As a "workaround", can you use a stacked bar chart?

**Comment by github-pba** [comment]
I don't think this would work, because the stacked bar chart seems to work on time series, and it always diplays all tags of the filtered cases. 

My feature request is not too important, more a (very) nice to hava, so if you are working on something similar, I'm totally fine with if it comes some day in the future. Our workaround is to use the REST API to grab the raw data and filter it afterwards.
